core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/IntelliJCleanupTest.scala

```diff
@@ -0,0 +1,76 @@
+package org.virtuslab.ideprobe.dependencies
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.virtuslab.ideprobe.config.{DependenciesConfig, IdeProbeConfig, IntelliJProvider, IntellijConfig, PathsConfig}
+import org.virtuslab.ideprobe.ide.intellij.IntelliJFactory
+
+import java.nio.file.Path
+
+@RunWith(classOf[JUnit4])
+final class IntelliJCleanupTest {
+  @Test
+  def existingIntelliJShouldNotBeDeletedDuringCleanup: Unit = givenInstalledIntelliJ { installationRoot =>
+      val intelliJProvider = IntelliJProvider.from(
+        IdeProbeConfig(
+          intellij = IntellijConfig.Existing(installationRoot, Seq.empty),
+          workspace = None,
+          resolvers = DependenciesConfig.Resolvers(),
+          driver = IntelliJFactory.Default.config,
+          paths = PathsConfig()
+        )
+      )
+      val existingInstalledIntelliJ = intelliJProvider.setup()
+
+      //when
+      existingInstalledIntelliJ.cleanup()
+
+      //then
+      assert(installationRoot.toFile.exists(), "The provided IntelliJ instance should exist after cleanup, but it was deleted.")
+    }
+
+  @Test
+  def existingIntelliJShouldRetainItsOriginalPluginsDuringCleanup: Unit = givenInstalledIntelliJ { installationRoot =>
+    //given a pre-installed IntelliJ and an IntelliJProvider
+    val preInstalledPlugins = installationRoot.resolve("plugins").toFile.list().toSet
+    val pluginsToInstall = Seq( //Plugins choice is arbitrary.
+      Plugin.Versioned("9667", "1.0.298", None),
+      Plugin.Versioned("8327", "2021.1-1.5.9", None)
+    )
+    val intelliJProvider = IntelliJProvider.from(
```

Comment:
I mentioned earlier to refactor creating this, but if it ends up with so many lines of repetitive code, I'd suggest to extract method.

Reply:
Will be done together with the factory/provider renaming.

Reply:
Switched to creating provider from config strings.